executorlib/base/executor.py

```
@@ -62,7 +62,7 @@ def future_queue(self) -> queue.Queue:
         """
         return self._future_queue
 
-    def submit(self, fn: callable, *args, resource_dict: dict = {}, **kwargs) -> Future:
+    def submit(self, fn: Callable, *args, resource_dict: dict = {}, **kwargs) -> Future:  # type: ignore
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)